app/models/miq_queue.rb

```diff
@@ -246,27 +250,44 @@ def unget(options = {})
     _log.info("#{MiqQueue.format_full_log_msg(self)}, Requeued")
   end
 
+  # TODO (juliancheal) This is a hack. Brakeman was giving us an SQL injection
+  # warning when we concatonated the queue_name string onto the query.
+  # Creating two seperate queries like this, resolves the Brakeman issue, but
+  # isn't idea. This will need to be rewritten using Arel queires at some point.
+
   MIQ_QUEUE_PEEK = <<-EOL
     state = 'ready'
     AND (zone IS NULL OR zone = ?)
+    AND (role IS NULL OR role IN (?))
+    AND (server_guid IS NULL OR server_guid = ?)
+    AND (deliver_on IS NULL OR deliver_on <= ?)
+    AND (priority <= ?)
     AND queue_name = ?
+  EOL
+
+  MIQ_QUEUE_PEEK_ARRAY = <<-EOL
+    state = 'ready'
+    AND (zone IS NULL OR zone = ?)
     AND (role IS NULL OR role IN (?))
     AND (server_guid IS NULL OR server_guid = ?)
     AND (deliver_on IS NULL OR deliver_on <= ?)
     AND (priority <= ?)
+    AND queue_name in (?)
```

Comment:
@kbrock is there a perf issue with always using IN? I think PG was smart enough to optimize that if it's e.g ```queue_name IN ('generic')```